spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteTablePathSparkAction.java

```diff
@@ -219,8 +219,10 @@ private String validateVersion(TableMetadata tableMetadata, String versionFileNa
       }
     }
 
-    Preconditions.checkNotNull(
```

Comment:
Just wondering, why this change is needed?  Isnt checkNotNull more specific?

Reply:
technically speaking, IAE is more specific since it tells that the user provided invalid arguments, which in this case is the `versionFileName`. See also my other comment in https://github.com/apache/iceberg/pull/11931#discussion_r1922479965 on this

Reply:
Sorry missed this conversation, makes sense to me.